### PR TITLE
CO-725: Meta XE API frontend application

### PR DESCRIPTION
**Pull request by magical**
Adapted from the OAuth2 consent demo app

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+FROM python:2.7-alpine
+MAINTAINER Andrew Ekstedt <ekstedta@oregonstate.edu>
```
Comment (jaredkosanovic) [inline Dockerfile]: Could you use [LABEL](https://docs.docker.com/engine/reference/builder/#label) here instead? https://docs.docker.com/engine/reference/builder/#maintainer-deprecated

File: README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
+Meta XE Frontend
```
Comment (jaredkosanovic) [inline README.md:1]: Could you add a config.py example file to this repo?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,21 @@
+FROM python:2.7-alpine
```
Comment (jaredkosanovic) [inline Dockerfile:1]: Could you add a [healthcheck](https://docs.docker.com/engine/reference/builder/#healthcheck) to this image?

File: metaxe/templates/index.html [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+                {% else %}
+                  <li class="next disabled"><a>Next <span aria-hidden="true">&rarr;</span></a></li>
+                {% endif %}
+              </ul>
+            </nav>
+          </div>
+        </div>
+      </div>
+    </section>
+
```
Comment (jaredkosanovic) [inline metaxe/templates/index.html:141]: Multiple blank lines

File: metaxe/templates/index.html [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+                      {{- instance -}}
+                    </option>
+                  {% endfor %}
+                </select>
+              </div>
+            </div>
+            <div class="col-sm-1">
+              <div class="form-group">
+                <label><br>
+                <button class="btn btn-primary">Filter</button>
```
Comment (jaredkosanovic) [inline metaxe/templates/index.html]: Maybe we should add a "clear" or "reset" button around this area to make the instance set to "Any" and clear the search box.

File: requirements.txt [hunk 6]
```diff
@@ -0,0 +1,20 @@
+appdirs==1.4.3
```
Comment (jaredkosanovic) [inline requirements.txt]: As you suggested, I had to change this to appdirs==1.4.2 or else I couldn't build the docker image. It told me that 1.4.2 was already installed (maybe along with a different package?). I didn't see any issues with running 1.4.2.

**Review by jaredkosanovic — CHANGES_REQUESTED** [review]
Overall great work. I tested this by running it in a container and it looks great and functions well.

File: Dockerfile (hunk 1 above)
Comment (magical) [reply Dockerfile]: Huh. When was maintainer deprecated?

**Comment by floppyears** [comment]
* When viewing the list of apps in any instance the distinction between the applications is not as easy on the eyes. Either adding some top & bottom margin or alternate color on the rows would help.

File: requirements.txt (hunk 6 above)
Comment (floppyears) [reply requirements.txt]: I ran into this issue as well.

File: Dockerfile (hunk 1 above)
Comment (jaredkosanovic) [reply Dockerfile]: @magical https://github.com/docker/docker/pull/25466

File: metaxe/templates/index.html [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
+    <!--[if lt IE 9]>
+      <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
+      <script src="https://oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
+    <![endif]-->
+  </head>
+
+  <body>
+    <header>
+      <a href="http://oregonstate.edu"><img id="logo" src="http://oregonstate.edu/images/logo.jpg?v=1" alt="Oregon State University" width="215" height="78"/></a>
```
Comment (floppyears) [inline metaxe/templates/index.html:28]: Why not specify the width and height in css?

Comment (magical) [reply metaxe/templates/index.html:28]: Those are from the OSU template I based this off of.

**Review by jaredkosanovic — APPROVED** [review]

**Pull request merged**